### PR TITLE
Add trendline options to scatterplot

### PR DESCRIPTION
**Pull request by roubkar**
- Added trendline options popover
- Added regression calculation methods for SLR and LOESS

### CODE REVIEW

**Comment by roubkar** [comment]
Resolves https://github.com/aimhubio/aim/issues/1160

**Review by arsengit — APPROVED** [review]

File: aim/web/ui/src/types/services/models/explorer/createAppModel.d.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ import { IRunsAppModelState } from 'types/services/models/runs/runsAppModel';
 
 import { ChartTypeEnum, CurveEnum } from 'utils/d3';
 import { SmoothingAlgorithmEnum } from 'utils/smoothingData';
+import { ITrendlineOptions } from '../scatter/scatterAppModel';
```
Comment (KaroMourad) [inline aim/web/ui/src/types/services/models/explorer/createAppModel.d.ts:24]: let's use an absolute path here

File: aim/web/ui/src/components/ScatterPlot/ScatterPlot.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import {
   getAxisScale,
   drawHoverAttributes,
 } from 'utils/d3';
+import drawScatterTrendline from 'utils/d3/drawScatterTrendline';
```
Comment (KaroMourad) [inline aim/web/ui/src/components/ScatterPlot/ScatterPlot.tsx:15]: let's export drawScatterTrendline function from utils/d3 index.ts as named export like other utils functions

**Review by KaroMourad — APPROVED** [review]

**Review by rubenaprikyan — APPROVED** [review]
lgtm

**Pull request merged**